### PR TITLE
Use ResourceType

### PR DESCRIPTION
**Pull request by GnomedDev**
Would have called it `ResourceChannelId`, but that is so long it causes the formatter to do line wrapping.

### CODE REVIEW

**Review by vabold — CHANGES_REQUESTED** [review]

File: payload/game/system/MultiDvdArchive.hh [hunk 1]
```diff
@@ -4,20 +4,20 @@
 
 namespace System {
 
+enum class ResChannelId {
+    Race = 0,
+    Course = 1,
+    Menu = 2,
+    Font = 3,
+};
+
```
[the remark this answers is not in the document]
Comment (GnomedDev) [reply payload/game/system/MultiDvdArchive.hh]: Resolved in ee0528969b6e3344899b46f81ec439fbaba62b42

**Comment by stblr** [comment]
What about `ResourceType`?

**Comment by GnomedDev** [comment]
Sounds like a saner name

**Comment by GnomedDev** [comment]
Done

**Review by riidefi — CHANGES_REQUESTED** [review]
Can we move the main definition out of MultiDvdArchive, and then have MultiDvdArchive::Type alias that main definition?

**Comment by GnomedDev** [comment]
Why?

**Comment by riidefi** [comment]
For instance in MK7, this is `System::EArchiveID`. I generally think its also consistent to have `UI::PageId` rather than `UI::Page::PageId` or `UI::Section::SectionId`, for instance.

**Comment by riidefi** [comment]
I also wouldn't object to moving this enum to `Registry`, even.

**Comment by GnomedDev** [comment]
Done

**Review by stblr — APPROVED** [review]

**Pull request merged**